File: server/models/product.js
const {DataTypes, Model} = require('sequelize');
const sequelize = require('../util/database');

class Product extends Model {
}

Product.init({
    // Model attributes are defined here
    productId: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true
    },
    productName: {
        type: DataTypes.STRING,
        allowNull: false
        // allowNull defaults to true
    },
    productDetails: {
        type: DataTypes.STRING,
        allowNull: true
    },
    createdByUserId: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    modifiedByUserId: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    statusId: {
        type: DataTypes.INTEGER,
        allowNull: true
    }
}, {
    createdAt: 'createdDate',
    updatedAt: `modifiedDate`,
    timestamps: true,
    // Other model options go here
    sequelize, // We need to pass the connection instance
    modelName: 'Product' // We need to choose the model name
});

// the defined model is the class itself
console.log(Product === sequelize.models.Product); // true

module.exports = Product;

File: succor/lib/models/product.dart
import 'dart:io';

class Product {
  int productId;
  String productName;
  String productDetails;
  String productImageUrl;
  File image;
  int createdByUserId;
  int modifiedByUserId;
  DateTime createdDate;
  DateTime modifiedDate;
  int statusId;

  Product();

  clone() => Product.fromJson(toJson());

  Product.fromJson(Map<String, dynamic> json)
      : productId = json['productId'],
        productName = json['productName'],
        productDetails = json['productDetails'],
        productImageUrl = json['productImageUrl'],
        createdByUserId = json['createdByUserId'],
        modifiedByUserId = json['modifiedByUserId'],
        createdDate = (json['createdDate'] != null ? DateTime.parse(json['createdDate']) : null),
        modifiedDate = (json['modifiedDate'] != null ? DateTime.parse(json['modifiedDate']) : null),
        statusId = json['statusId'];

  Map<String, dynamic> toJson() => {
        'productId': productId,
        'productName': productName,
        'productDetails': productDetails,
        'productImageUrl': productImageUrl,
        if (createdDate != null) 'createdDate': createdDate.toString(),
        if (modifiedDate != null) 'modifiedDate': modifiedDate.toString(),
        'createdByUserId': createdByUserId,
        'modifiedByUserId': modifiedByUserId,
        if (statusId != null) 'statusId': statusId,
      };
}
